### PR TITLE
Add support for conduit token being configured globally

### LINKED ISSUE #18: Manage Conduit token within Jenkins UI
Currently, this plugin relies on the `~/.arcrc` file. Instead, it would be better to allow the Conduit token to be managed in the Jenkins UI and be passed to arcanist with the `--conduit-token` flag.

### PR DESCRIPTION
**Pull request by sectioneight**
Fixes #18

### CODE REVIEW

**Comment by sectioneight** [comment]
Tested manually my renaming my ~/.arcrc, setting up a conduit token in the global configuration and running a job. Note that the conduit token is clearly visible in the job output, so if this is a security concern we will want to figure out a way to scrub.

File: src/main/java/com/uber/jenkins/phabricator/PhabricatorBuildWrapper.java [hunk 1]
```diff
@@ -54,12 +54,17 @@ public Environment setUp(AbstractBuild build,
...
 
+        final String conduitToken = this.getConduitToken();
+        if (!CommonUtils.isBlank(conduitToken)) {
+            // Make this available for the post-build action (PhabricatorNotifier.class)
+            envAdditions.put(PhabricatorPlugin.CONDUIT_TOKEN, conduitToken);
+        }
+
         String diffID = environment.get(PhabricatorPlugin.DIFFERENTIAL_ID_FIELD);
-        if (diffID == null || "".equals(diffID)) {
+        if (CommonUtils.isBlank(diffID)) {
```
Comment (sectioneight) [inline src/main/java/com/uber/jenkins/phabricator/PhabricatorBuildWrapper.java]: unrelated cleanup

**Comment by sectioneight** [comment]
@joshuaspence @jjx

File: src/main/java/com/uber/jenkins/phabricator/conduit/Differential.java [hunk 2]
```diff
@@ -36,7 +36,7 @@
 public class Differential {
     private final JSONObject rawJSON;
     private final LauncherFactory launcher;
-    private final String diffID;
```
Comment (sectioneight) [inline src/main/java/com/uber/jenkins/phabricator/conduit/Differential.java]: this field was unused

File: src/main/java/com/uber/jenkins/phabricator/PhabricatorBuildWrapper.java [hunk 3]
```diff
@@ -114,6 +119,10 @@ public Environment setUp(AbstractBuild build,
             if (!createCommit) {
                 patchCommand.add("--nocommit");
             }
+            // Use a globally-configured conduit token, if applicable
+            if (!CommonUtils.isBlank(conduitToken)) {
+                patchCommand.add("--conduit-token=" + this.getConduitToken());
```
Comment (jjx) [inline src/main/java/com/uber/jenkins/phabricator/PhabricatorBuildWrapper.java]: Will this get rendered as plain text to stdout? If it does, any concerns there?

Comment (sectioneight) [inline src/main/java/com/uber/jenkins/phabricator/PhabricatorBuildWrapper.java]: Yea, I left that as a comment on the overall PR. We probably want to filter this out, although in general I think it's useful to have the commands being run being displayed. @joshuaspence?

File: src/main/java/com/uber/jenkins/phabricator/conduit/ArcanistClient.java [hunk 4]
```diff
@@ -35,16 +36,29 @@
...
+            sb.append(this.conduitToken);
+        }
+        return sb.toString();
     }
 
     public JSONObject callConduit(Launcher.ProcStarter starter, PrintStream stderr) throws IOException, InterruptedException {
         JSONObject obj = new JSONObject();
         obj.putAll(this.params);
-        List<String> command = new ArrayList<String>(Arrays.asList("sh", "-c", "echo '" + obj.toString() + "' | arc call-conduit " + this.methodName));
+        List<String> command = new ArrayList<String>(
```
Comment (jjx) [inline src/main/java/com/uber/jenkins/phabricator/conduit/ArcanistClient.java]: Arrays.asList returns a list, no need to create a new ArrayList using the List

Comment (sectioneight) [inline src/main/java/com/uber/jenkins/phabricator/conduit/ArcanistClient.java]: Done. Good catch.

File: src/main/java/com/uber/jenkins/phabricator/PhabricatorBuildWrapper.java (hunk 3 above)
Comment (sectioneight) [inline src/main/java/com/uber/jenkins/phabricator/PhabricatorBuildWrapper.java]: The head commit on this branch now suppresses the command name, which alleviates any security concerns but makes debugging harder. Not sure how I feel about it.

**Comment by jjx** [comment]
lgtm
